File: src/utils/resolveParenthesizedType.ts
import ts, {TypeNode} from "typescript";

export function resolveParenthesizedType(node: TypeNode): TypeNode {
    if (ts.isParenthesizedTypeNode(node)) {
        return resolveParenthesizedType(node.type)
    } else {
        return node
    }
}

File: gradle-plugin/karakum/src/jsMain/kotlin/io/github/sgrishchenko/karakum/util/resolveParenthesizedType.kt
package io.github.sgrishchenko.karakum.util

import typescript.TypeNode
import typescript.isParenthesizedTypeNode

fun resolveParenthesizedType(node: TypeNode): TypeNode {
    return if (isParenthesizedTypeNode(node)) {
        resolveParenthesizedType(node.type)
    } else {
        node
    }
}
